pkg/streamingpromql/types/ring_buffer_test.go

```diff
@@ -13,26 +13,30 @@ import (
 	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
 )
 
-// We want to ensure fPoint+hPoint ring buffers are tested consistently,
+// We want to ensure FPoint+HPoint ring buffers are tested consistently,
 // and we don't care about performance here so we can use an interface+generics.
 type ringBuffer[T any] interface {
 	DiscardPointsBefore(t int64)
 	Append(p T) error
-	ForEach(f func(p T))
-	UnsafePoints(maxT int64) (head []T, tail []T)
-	CopyPoints(maxT int64) ([]T, error)
-	LastAtOrBefore(maxT int64) (T, bool)
-	CountAtOrBefore(maxT int64) int
-	AnyAtOrBefore(maxT int64) bool
-	First() T
 	Reset()
 	Use(s []T)
 	Release()
+	ViewUntilForTesting(maxT int64, searchForwards bool) ringBufferView[T]
```

Comment:
I know this comes from an earlier commit in the history, but I think you should drop `ViewUntilForTesting` and implement a ViewUntilSearchingForwards/Backwards

Reply:
Done in f3ac4c3